app/src/userpages/components/StreamPage/Show/ConfigureView/NewFieldEditor/index.jsx

```diff
@@ -0,0 +1,121 @@
+// @flow
+
+import React, { Component } from 'react'
+import { Col, Row, Button } from 'reactstrap'
+import { I18n, Translate } from 'react-redux-i18n'
+
+import Dropdown from '$shared/components/Dropdown'
+import TextInput from '$shared/components/TextInput'
+import type { StreamField } from '$shared/flowtype/stream-types'
+
+import { leftColumn, rightColumn, fieldTypes } from '../../../constants'
+import styles from './newFieldEditor.pcss'
+
+type Props = {
+    previousFields: Array<StreamField>,
+    onConfirm: (name: string, type: string) => void,
+    onCancel: () => void,
+}
+
+type State = {
+    name: string,
+    nameError: ?string,
+    type: string,
+}
+
+export class NewFieldEditor extends Component<Props, State> {
+    state = {
+        name: '',
+        nameError: null,
+        type: fieldTypes[0],
+    }
+
+    onNameChange = (value: string) => {
+        this.setState({
+            name: value,
+        }, () => this.validate())
+    }
+
+    onTypeChange = (value: string) => {
+        this.setState({
+            type: value,
+        })
+    }
+
+    validate = () => {
+        const { name } = this.state
+        const { previousFields } = this.props
+        let error = null
+
+        if (name.length === 0) {
+            error = I18n.t('userpages.streams.edit.configure.newFieldEditor.error.emptyName')
+        }
+        if (previousFields.find((field) => field.name === name)) {
```

Comment:
`error` is overwritten here if both conditions are true.

Reply:
Yes, field.name should never be empty. If it is, then something is wrong with the backend. Also I think it's correct behaviour to complain about duplicate field name if this happens :)